rest/data/build_baron.go

```diff
@@ -103,8 +103,17 @@ func fileTicketCustomHook(context context.Context, taskId string, execution int,
 	return resp, nil
 }
 
-func makeNotification(settings *evergreen.Settings, project string, t *task.Task) (*notification.Notification, error) {
-	payload, err := trigger.JIRATaskPayload("", project, settings.Ui.Url, "", "", t)
+func makeNotification(ctx context.Context, settings *evergreen.Settings, project string, t *task.Task) (*notification.Notification, error) {
+	mappings := &evergreen.JIRANotificationsConfig{}
```

Comment:
the mappings fetch was happening further down the stack in [JIRATaskPayload](https://github.com/evergreen-ci/evergreen/blob/45bcc509fe7165d2dd549727274e38f6b7cff24a/trigger/task.go#L966-L968). Doing it earlier made it possible to avoid threading the context through the task trigger event handlers and instead to do it in [Fetch](https://github.com/evergreen-ci/evergreen/blob/3ede2d9ecf509f4c3e95a7e3479a6df1bd2dc55c/trigger/task.go#L160). The build baron notification was another thing that used the same JIRATaskPayload function, so it also had to get the mapping.